src/core/Meganav.tsx

```diff
@@ -107,147 +23,84 @@ export type MeganavNoticeProps = {
-  loginLink = "/login",
-  urlBase,
-  addSearchApiKey,
-  statusUrl,
-  searchDataId,
-}: MeganavProps) => {
-  const [sessionState, setSessionState] = useState<MeganavSessionState>();
 
   useEffect(() => {
-    // Note if state is never updated, sessionState stays null and never removes the placeholder.
-    // This makes SSR consistent (ie. we always show the placeholder)
-    connectState(selectSessionData, setSessionState);
+    const observeNoticeResize = () => {
+      const noticeElement = document.querySelector('[data-id="ui-notice"]');
+      if (noticeElement) {
+        setNoticeHeight(noticeElement.getBoundingClientRect().height);
+      }
+    };
+    observeNoticeResize();
+    window.addEventListener("resize", observeNoticeResize);
+    return () => window.removeEventListener("resize", observeNoticeResize);
   }, []);
 
-  useEffect(() => {
-    const teardown = MeganavScripts({ themeName, addSearchApiKey });
-    return () => teardown();
-  }, [sessionState]);
-
-  const theme = MeganavData.themes[themeName] as MeganavTheme;
-  const absUrl = (path: string) => _absUrl(path, urlBase);
-
   return (
-    <nav
-      className={`ui-meganav-wrapper ${theme.backgroundColor} ${theme.barShadow}`}
-      data-id="meganav"
-      aria-label="Main"
-    >
-      {notice && <Notice {...notice.props} config={notice.config} />}
-      <div className="ui-meganav ui-grid-px">
-        <div className="mr-24">
-          <Logo dataId="meganav-logo" href={urlBase} logoUrl={paths?.logo} />
-        </div>
-
-        <MeganavItemsDesktop
-          panels={panels}
-          paths={paths}
-          theme={theme}
-          absUrl={absUrl}
-          statusUrl={statusUrl}
-        />
-
-        {/* Because we load the session state through fetch, we display a placeholder until fetch returns */}
-        {sessionState ? (
-          <SignIn
-            sessionState={sessionState}
-            theme={theme}
-            loginLink={loginLink}
-            absUrl={absUrl}
-            searchDataId={searchDataId}
-          />
-        ) : (
-          <SignInPlaceholder />
-        )}
-
-        <MeganavItemsMobile
-          panels={panels}
+    <>
+      <div
+        className="absolute inset-0 w-full z-50"
+        id={theme === "dark" ? "meganav-theme-dark" : "meganav"}
+        data-testid="meganav"
+        style={{ height: HEADER_HEIGHT + noticeHeight }}
+      >
+        {notice && <Notice {...notice.props} config={notice.config} />}
+        <Header
+          className="max-w-screen-xl mx-auto px-0 sm:px-32 md:px-40 lg:px-64"
+          isNoticeVisible={!!notice}
+          nav={
+            <Flyout
+              menuItems={menuItemsForHeader}
+              className="justify-left z-40"
+              flyOutClassName="flex justify-left"
+              viewPortClassName="ui-shadow-lg-medium border border-neutral-000 dark:border-neutral-1300 rounded-2xl mt-8 bg-neutral-000 dark:bg-neutral-1300"
+              hasAnimation={true}
```

Comment:
Are there situations where we wouldn't want the animation? We can probably remove this prop

Reply:
no yes, will remove it